src/samples.jl

```diff
@@ -80,6 +80,10 @@ function Base.hash(a::Samples, h::UInt)
     return hash(Samples, hash(a.encoded, hash(a.info, hash(a.data, h))))
 end
 
+function Base.convert(::Type{Samples{T}}, samples::Samples) where {T}
+    return Samples(convert(T, samples.data), samples.info, samples.encoded)
+end
```

Comment:
I don't think we want to convert encoded samples like this...I think you can easily get into trouble if you for instance convert Int8 encoded samples to Float32 adn then save them with `store`, you'll get totally wrong Signal (`store` just calls `write_lpcm` with `encode(samples).data` https://github.com/beacon-biosignals/Onda.jl/blob/97446538cedb2017d575097cc128eeea8755bd52/src/samples.jl#L519 and `encode` does nothing if `samples.encoded` is true).
```suggestion
function Base.convert(::Type{Samples{T}}, samples::Samples) where {T}
    samples.encoded && throw(ArgumentError("can't `convert` encoded samples; use `decode` first"))
    return Samples(convert(T, samples.data), samples.info, samples.encoded)
end
```

Reply:
I'm fine with fully disallowing this, but I did test that validation will catch this issue:
```julia
    # For encoded samples, in generally we cannot `convert`.
    # We choose to not update encoding parameter in `convert`, since `convert` can be implied
    # implicitly, and changing the encoding parameters seems like too big of a change.
    # Therefore, validation errors.
    samples = Samples(rand(sample_type(info), 3, 100), info, true)
    err = ArgumentError("encoded `data` matrix eltype does not match `sample_type(info)`")
    @test_throws err convert(Samples{Matrix{Int32}}, samples)
```

Reply:
ahh right, the validation in the constructor, forgot about that.  I guess there's very little usefulness in using convert on encoded since you'd have to already have samples with a _different_ eltype and you're converting to the correct type for it to pass validation...